app/views/returnFrequency/ChooseDates.scala.html

```diff
@@ -15,61 +15,92 @@
 @import models.returnFrequency._
 @import models.auth.User
 @import views.html.MainTemplate
 @import uk.gov.hmrc.play.views.html.helpers.FormWithCSRF
+@import config.AppConfig
 
-@this(mainTemplate: MainTemplate, errorSummary: ErrorSummary, radioGroup: RadioGroup, helpersForm: FormWithCSRF)
+@this(mainTemplate: MainTemplate,
+      govukBackLink: GovukBackLink,
+      govukRadios: GovukRadios,
+      govukErrorSummary: GovukErrorSummary,
+      govukButton: GovukButton,
+      formWithCSRF: FormWithCSRF)
 
-@(form: Form[ReturnDatesModel], current: ReturnPeriod)(implicit user : User[_], messages: Messages, appConfig: config.AppConfig)
+@(form: Form[ReturnDatesModel], current: ReturnPeriod)(implicit user : User[_], messages: Messages, appConfig: AppConfig)
 
-
-@additionalContent = {
-    @if(current.id.equals("Annually")){
-        <p id="form-hint" class="form-hint">
-            <span id="currently-set-text">@messages("return_frequency.currentAnnual")</span>
-        </p>
+@formHint = @{
+    if(current.id.equals("Annually")){
+      messages("return_frequency.currentAnnual")
     } else {
-        <p id="form-hint" class="form-hint">
-            <span id="currently-set-text">@messages("return_frequency.current")</span>
-            <span id="currently-set-period">@messages(s"return_frequency.${current.id}")@messages("common.fullStop")</span>
-        </p>
+      messages("return_frequency.current") + " " +
+      messages(s"return_frequency.${current.id}") + messages("common.fullStop")
     }
 }
 
-@backLink = {
-  <a class="link-back" href='@appConfig.manageVatUrl'>@messages("base.back")</a>
+@availableOptions = @{Seq(Jan, Feb, Mar, Monthly).filterNot(_ == current)}
+
+@conditionalRevealHtml(value: String) = {
+    <p class="govuk-hint">
+        @(current, value) match {
+            case (Monthly, _) | (_, "Monthly") => {@messages("return_frequency.changeAtEndOfMonth")}
+            case _ => {@messages("return_frequency.changeAtEndOfQuarter")}
+        }
+    </p>
 }
 
 @mainTemplate(
-  title = if(form.errors.nonEmpty) messages("common.error.prefixTitle", messages("return_frequency.title")) else messages("return_frequency.title"),
-  appConfig = appConfig,
-  backLinkContent = Some(backLink),
+  pageTitle =
+    if(form.errors.nonEmpty) messages("common.error.prefixTitle",
+    messages("return_frequency.title")) else messages("return_frequency.title"),
   user = Some(user)) {
 
-  @errorSummary("common.errorSummary.heading", form)
+  @govukBackLink(BackLink(
+    href = appConfig.manageVatUrl,
+    content = Text(messages("base.back"))
+  ))
 
-  @helpersForm(action = controllers.returnFrequency.routes.ChooseDatesController.submit()) {
+  @if(form.hasErrors) {
+    @govukErrorSummary(ErrorSummary(
+        title = Text(messages("common.errorSummary.heading")),
+        errorList = form.errors.map { error =>
+          ErrorLink(
+            href = availableOptions.lift(0).map(option => s"#return_frequency.${option.id}"),
```

Comment:
The link in the error summary doesn't go anywhere. I think this can be resolved by simply setting this href to `"period-option"` as that will always be the ID of the first radio item